### PR TITLE
Json fix

### PR DESCRIPTION
**Pull request by ryansurf**
Hello @K-dash !

This PR mostly cleans up `helper.py`. Addresses https://github.com/ryansurf/cli-surf/issues/84

# Changes

### `src/api.py`

Mostly organizing code. `forecast()` now returns a dictionary, not several nested lists, for reusability 

### `src/helper.py`

- This file has too many `if` statements. These were replaced by dictionary mappings, which makes the code look cleaner.
- ` forecast_to_json` was altered to take a dictionary as input, from the `forecast()` function in `src/api.py`

### `src/test_api.py`

`test_forecast()` was failing because of the dictionary now being returned from the forecast. Fixed this

### `src/test_helper.py`

A new unit test was added to test the json argument. Prior to this, the json argument was causing an error and we were non-the-wiser because our tests didn't catch this!

## Summary by Sourcery

Refactor helper functions to use mappings for setting and printing argument values, update forecast data handling to use dictionaries, and enhance test coverage to validate these changes.

Enhancements:
- Refactor `set_output_values` to use a mapping dictionary for setting argument values.
- Refactor `print_ocean_data` and `print_forecast` to use mapping lists for printing data.
- Update `forecast_to_json` to handle forecast data as a dictionary instead of a list.
- Improve `print_outputs` function to handle ocean data dictionary more effectively.

Tests:
- Add `test_json_output` to verify JSON output structure and data types.
- Update `test_forecast` to validate the new dictionary-based forecast data structure.

### CODE REVIEW

**Comment by K-dash** [comment]
@ryansurf 
I'll be reviewing it tonight, so hold on a bit!

**Comment by ryansurf** [comment]
> @ryansurf I'll be reviewing it tonight, so hold on a bit!

No rush!

File: src/helper.py [hunk 1]
```diff
@@ -44,52 +46,63 @@ def arguments_dictionary(lat, long, city, args):
...
-    if "show_rain_sum" in args or "srs" in args:
-        arguments["show_rain_sum"] = 1
-    if "show_precipitation_prob" in args or "spp" in args:
-        arguments["show_precipitation_prob"] = 1
+    and sets the appropritate values in the
+    arguments_dictionary(show_wave = 1, etc).
+    Returns the arguments_dictionary dict with the updated CLI args
+    """
+    # map of arguments to dictionary keys & values
+    mappings = {
```
Comment (K-dash) [inline src/helper.py:57]: Great job on using a dictionary to eliminate if-statement branching - it's made the code much cleaner!
However, I believe we can further improve maintainability and extensibility by managing the Arguments and Mapping values using model classes.
For example, we could implement it like this:
```python
# arguments.py
from typing import Literal, Dict, Any
from pydantic import BaseModel, Field

class Arguments(BaseModel):
    lat: float
    long: float
    city: str
    show_wave: bool = True
    show_large_wave: bool = False
    show_uv: bool = True
    show_height: bool = True
    show_direction: bool = True
    show_period: bool = True
    show_city: bool = True
    show_date: bool = True
    show_air_temp: bool = False
    show_wind_speed: bool = False
    show_wind_direction: bool = False
    json_output: bool = False
    show_rain_sum: bool = False
    show_precipitation_prob: bool = False
    unit: Literal["imperial", "metric"] = "imperial"
    decimal: int = Field(default=1, ge=0)
    forecast_days: int = Field(default=0, ge=0, le=7)
    color: str = "blue"
    gpt: bool = False

class ArgumentMappings(BaseModel):
    show_wave: bool = Field(default=True, description="Show wave information", aliases=["sw", "hide_wave", "hw"])
    show_large_wave: bool = Field(default=False, description="Show large wave information", aliases=["slw"])
    show_uv: bool = Field(default=True, description="Show UV information", aliases=["suv", "hide_uv", "huv"])
    show_height: bool = Field(default=True, description="Show wave height information", aliases=["sh", "hide_height", "hh"])
    show_direction: bool = Field(default=True, description="Show wave direction information", aliases=["sd", "hide_direction", "hdir"])
    show_period: bool = Field(default=True, description="Show wave period information", aliases=["sp", "hide_period", "hp"])
    show_city: bool = Field(default=True, description="Show city information", aliases=["sc", "hide_location", "hl"])
    show_date: bool = Field(default=True, description="Show date information", aliases=["sdate", "hide_date", "hdate"])
    unit: Literal["imperial", "metric"] = Field(default="imperial", description="Set unit system", aliases=["m", "metric"])
    json_output: bool = Field(default=False, description="Enable JSON output", aliases=["j", "json"])
    gpt: bool = Field(default=False, description="Enable GPT functionality", aliases=["g"])
    show_air_temp: bool = Field(default=False, description="Show air temperature", aliases=["sat"])
    show_wind_speed: bool = Field(default=False, description="Show wind speed", aliases=["sws"])
    show_wind_direction: bool = Field(default=False, description="Show wind direction", aliases=["swd"])
    show_rain_sum: bool = Field(default=False, description="Show rain sum", aliases=["srs"])
    show_precipitation_prob: bool = Field(default=False, description="Show precipitation probability", aliases=["spp"])

    @classmethod
    def generate_mappings(cls) -> Dict[str, str]:
        mappings = {}
        for field_name, field in cls.model_fields.items():
            mappings[field_name] = field_name
            for alias in field.alias_priority or []:
                mappings[alias] = field_name
        return mappings

    class Config:
        allow_population_by_field_name = True
```
```python
# Excerpts from helper.py
from argument_types import Arguments, ArgumentMappings
from typing import List

def arguments_dictionary(lat: float, long: float, city: str, args: List[str]) -> Arguments:
    """
    Creates and returns an Arguments object with cli argument values.
    The object is initialized with the given lat, long, and city,
    then updated based on the provided command line arguments.
    """
    initial_args = {
        "lat": lat,
        "long": long,
        "city": city,
        "decimal": extract_decimal(args),
        "forecast_days": get_forecast_days(args),
        "color": get_color(args),
    }
    arguments = Arguments(**initial_args)
    
    arg_mappings = ArgumentMappings()
    argument_mapping_dict = ArgumentMappings.generate_mappings()
    
    for arg in args:
        if arg in argument_mapping_dict:
            target_field = argument_mapping_dict[arg]
            if arg.startswith("hide_") or arg in ["hw", "huv", "hh", "hdir", "hp", "hl", "hdate"]:
                setattr(arguments, target_field, False)
            else:
                value = getattr(arg_mappings, target_field)
                setattr(arguments, target_field, value)
    
    return arguments
```
(This is just an example and could be implemented more cleanly.)

What do you think about addressing these changes in the next refactoring phase, perhaps as a separate PR? This way, we can build upon the current improvements and further enhance our code structure.

This approach would allow us to:
- Leverage type safety and validation provided by model classes
- Centralize our argument definitions, making it easier to add or modify them
- Improve IDE support and autocompletion

If you agree, we can create a new issue to track this potential improvement😎

**Review by K-dash — COMMENTED** [review]
As sourcery-ai suggests, you might want to add a test if there is a possibility of an edge case.

File: src/helper.py (hunk 1 above)
Comment (ryansurf) [reply src/helper.py:57]: This would be great! I like the addition of classes/typing/pydantic, it makes the code much easier to read/maintain. Lets create the new issue to work on this

**Review by K-dash — APPROVED** [review]
